crates/interpreter/src/instructions/bitwise.rs

```diff
@@ -71,29 +70,35 @@ pub fn byte<H: Host + ?Sized>(interpreter: &mut Interpreter, _host: &mut H) {
     gas!(interpreter, gas::VERYLOW);
     pop_top!(interpreter, op1, op2);
 
-    let o1 = as_usize_saturated!(op1);
-    *op2 = if o1 < 32 {
-        // `31 - o1` because `byte` returns LE, while we want BE
-        U256::from(op2.byte(31 - o1))
-    } else {
-        U256::ZERO
-    };
+    let o1 = as_usize_saturated!(op1) % 32;
+    let byte_value = op2.byte(31 - o1);
+    *op2 = U256::from(byte_value);
```

Comment:
I don't think this is right, but also not sure if there is a test for this in statetests

Reply:
This way when 'o1' exceeds 31, the function returns ZERO but to ensure the implementation works as expected, can you suggest any specific tests we should do just in case?

Reply:
I don't think so? 33 % 32 = 1 so it will be 31 - 1 byte instead of zero

Reply:
Yes, you're right. Sometimes we have things right in front of our eyes and they escape us. If you have any ideas on how to improve the method, please let me know.

Reply:
we can shifts the op2 right by calculated bits to align and isolate the desired byte using bitmask but I dont know if the result is better than byte(..)

something like this:

```rs
*op2 = if o1 < 32 {
      let shift = (31 - o1) * 8;
      U256::from((*op2 >> shift) & U256::from(0xFF))
  } else {
      U256::ZERO
  };
```

What do you think ?

Reply:
I believe `byte` is already optimal on since it will simply index into the value as a byte array. Thanks for the efforts anyway.

Reply:
Thanks, I'm going to leave this method as it was =)